test/testlocaldiscovery.cpp

```diff
@@ -17,6 +17,87 @@ class TestLocalDiscovery : public QObject
     Q_OBJECT
 
 private slots:
+    void testSelectiveSyncQuotaExceededDataLoss()
+    {
+        FakeFolder fakeFolder{FileInfo{}};
+
+        // folders that fit the quota
+        fakeFolder.localModifier().mkdir("big-files");
+        fakeFolder.localModifier().insert("big-files/bigfile_A.data", 1000);
+        fakeFolder.localModifier().insert("big-files/bigfile_B.data", 1000);
+        fakeFolder.localModifier().insert("big-files/bigfile_C.data", 1000);
+        fakeFolder.localModifier().mkdir("more-big-files");
+        fakeFolder.localModifier().insert("more-big-files/bigfile_A.data", 1000);
+        fakeFolder.localModifier().insert("more-big-files/bigfile_B.data", 1000);
+        fakeFolder.localModifier().insert("more-big-files/bigfile_C.data", 1000);
+        QVERIFY(fakeFolder.syncOnce());
+        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        // folders that won't fit
+        fakeFolder.localModifier().mkdir("big-files-wont-fit");
+        fakeFolder.localModifier().insert("big-files-wont-fit/bigfile_A.data", 800);
+        fakeFolder.localModifier().insert("big-files-wont-fit/bigfile_B.data", 800);
+        fakeFolder.localModifier().mkdir("more-big-files-wont-fit");
+        fakeFolder.localModifier().insert("more-big-files-wont-fit/bigfile_A.data", 800);
+        fakeFolder.localModifier().insert("more-big-files-wont-fit/bigfile_B.data", 800);
+
+        const auto remoteQuota = 600;
+        QObject parent;
+        fakeFolder.setServerOverride([&](QNetworkAccessManager::Operation op, const QNetworkRequest &request, QIODevice *outgoingData) -> QNetworkReply * {
+            Q_UNUSED(outgoingData)
+            if (op == QNetworkAccessManager::PutOperation) {
+                if (request.rawHeader("OC-Total-Length").toInt() > remoteQuota) {
+                    return new FakeErrorReply(op, request, &parent, 507);
```

Comment:
you can directly use `fakeFolder`, no need to create an extra parent object

Reply:
@mgallien no, `fakeFolder `doesn't work, it is not `QObject`